packages/jest-config/package.json

```diff
@@ -18,12 +18,16 @@
   },
   "peerDependencies": {
     "@types/node": "*",
+    "esbuild-register": ">=3.1.0",
```

Comment:
I wonder if we could resolve from the context if the config file and avoid the peer dep?

Reply:
I'm not sure I follow what you mean here. My view was that this was documenting optional dependencies which can add additional functionality for jest-config, but I think could just totally remove it without it affecting the code at all.

Reply:
Taking a step back, we need the peer dep so that we can do `import(moduleSpecified)` and it resolves correctly (especially when using pnp or pnpm). However, if we resolve the module specified from the context of the config, we wouldn't need to specify the peer dep.

However, since I wanna change this in a future major so that we don't have to hard code support for modules in Jest, I think the current approach is fine for now 👍